File: backend/app/schemas/company.py
"""Pydantic schemas for Company I/O."""
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CompanyBase(BaseModel):
    ticker: str = Field(..., max_length=10)
    cik: str = Field(..., max_length=20)
    name: str = Field(..., max_length=255)
    sector: str | None = Field(default=None, max_length=100)
    industry: str | None = Field(default=None, max_length=100)
    market_cap: int | None = None


class CompanyCreate(CompanyBase):
    pass


class CompanyResponse(CompanyBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CompanyListResponse(BaseModel):
    companies: list[CompanyResponse]
    total: int

File: frontend/src/types/company.ts
export interface Company {
  id: string;
  ticker: string;
  cik: string;
  name: string;
  sector: string | null;
  industry: string | null;
  market_cap: number | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyListResponse {
  companies: Company[];
  total: number;
}

export interface Filing {
  id: string;
  company_id: string;
  filing_type: string;
  accession_number: string;
  filed_date: string;
  period_of_report: string | null;
  edgar_url: string | null;
  processed: boolean;
  created_at: string;
}

export interface FilingListResponse {
  filings: Filing[];
  total: number;
}

export interface JobStatus {
  id: string;
  job_type: string;
  status: "queued" | "running" | "completed" | "failed";
  payload?: Record<string, unknown> | null;
  result?: Record<string, unknown> | null;
  error?: string | null;
}
